### PR TITLE
Added lots of QOL in the code and added and API

### PR DESCRIPTION
**Pull request by TurtleGod7**
In this update, i added lots of QOL changes (like refactoring code) and I fixed some issues in the code. Also, I added a weather API and a new calculation based on that.

## Summary by CodeRabbit

- **New Features**
	- Enhanced main application functionality with new modules, improved menu options, and refined calculations for gas consumption.
	- Added weather functionality, including interaction with a weather API, user location preferences, and asynchronous weather data retrieval.
	- Implemented quality of life improvements across the application.

- **Documentation**
	- Updated the README with clearer instructions, revised sections for better clarity, and corrected typos.

- **Bug Fixes**
	- Fixed a typo in the Credits section of the README.

- **Refactor**
	- Refactored imports for better organization in the main application.

- **Chores**
	- Updated requirements with the addition of `python-weather==1.1.2`.

### CODE REVIEW

**Review by TurtleGod7 — COMMENTED** [review]
We need to edit the part of the readme to reflect over both measurment units

File: README.md [hunk 1]
```diff
@@ -1,45 +1,43 @@
...
 Thank You whoever you are for using my program!
 This code was started in January 2024. Originally I thought that this was a fun project and I hope that the work I have put in becomes great!
 Originally I wanted it to be some sort of Knock Off Google Maps but using a graph.
 This is my first time creating code for others, and I would love it if you guys give criticism.
 
-# Requirements:
+## Requirements
 
-There are 5 modules that need to be installed from PyPl for this program to work. To install these, you can either type into your command line (Windows):
+There are 5 modules that need to be installed from PyPl for this program to work. To install these, you can either type into your command line:
```
[the remark this answers is not in the document]
Comment (TurtleGod7) [reply README.md:10]: Done

File: README.md [hunk 2]
```diff
@@ -1,45 +1,43 @@
...
+### Inbuilt Modules
+
 - [Time](https://docs.python.org/3/library/time.html)
 - [Sys](https://docs.python.org/3/library/sys.html#module-sys)
 - [OS](https://docs.python.org/3/library/os.html#module-os)
 - [Sqlite3](https://docs.python.org/3/library/sqlite3.html#module-sqlite3)
 - [RE](https://docs.python.org/3/library/re.html#module-re)
 
-## PyPi Modules
+### PyPi Modules
```
[the remark this answers is not in the document]
Comment (TurtleGod7) [reply README.md:40]: Done

File: README.md [hunk 3]
```diff
@@ -1,45 +1,43 @@
...
 
 - [BeautifulSoup](https://pypi.org/project/beautifulsoup4/)
 - [Prettytable](https://pypi.org/project/prettytable/)
```
[the remark this answers is not in the document]
Comment (TurtleGod7) [reply README.md:43]: Done

File: README.md [hunk 4]
```diff
@@ -48,8 +46,8 @@ Thank you to the creators of:
 - [Termcolor](https://pypi.org/project/termcolor/)
 
 Because of their work, the code would have been a lot harder to make.
-Also, thanks to [@Magnus-Thomsen](https://github.com/Magnus-Thomsen) for the SQL database and I have hit GitHub linked.
+Also, thanks to [@Magnus-Thomsen](https://github.com/Magnus-Thomsen) for the SQL database and I have his GitHub linked.
 
-Note: This is my first ever version of releasing code on Github and any suggestions from passerbyers would be helpful!
+Note: This is my first ever time releasing code on Github and any suggestions from passerbyers would be helpful!
```
[the remark this answers is not in the document]
Comment (TurtleGod7) [reply README.md:51]: Done

File: main.py [hunk 5]
```diff
@@ -207,135 +189,124 @@ def add():
...
+    elif choice == 4:
+        up = 'cartype'
+        new = input("What is the updated car type?\n")
+    elif choice == 5:
+        up = 'isLeasingCar'
+        leasing_input = input("Is the car leased? (Yes/No)\n").lower()
+        new = 1 if leasing_input == 'yes' else 0
+
+    sqlUpdate = '''UPDATE cars SET '''+up+''' = ? WHERE id = ?'''
+    cursor.execute(sqlUpdate, (new, carID))
```
[the remark this answers is not in the document]
Comment (TurtleGod7) [reply main.py:302]: Although that may be true, this program is too small to be caring about SQL injections.

**Review by ghost — APPROVED** [review]
Its good 👍

**Pull request merged**